packages/editor/src/components/editor-help/test/index.native.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, render, waitForElementToBeRemoved } from 'test/helpers';
+
+/**
+ * Internal dependencies
+ */
+import EditorHelp from '../index';
+
+it( 'lists help topics', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+
+	expect( helpTopic ).toBeTruthy();
+} );
+
+it( 'navigates to help topic detail screen', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+	fireEvent.press( helpTopic );
+
+	expect(
+		screen.getByText(
+			'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+		)
+	).toBeTruthy();
+} );
+
+it( 'navigates back from help topic detail screen', async () => {
+	const screen = render( <EditorHelp isVisible /> );
+	const helpTopic = await screen.findByText( 'Customize blocks' );
+	fireEvent.press( helpTopic );
+
+	const detailScreenTitle1 = screen.getByText(
+		'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+	);
+	expect( detailScreenTitle1 ).toBeTruthy();
+
+	const backButton = screen.getAllByA11yLabel( 'Go back' );
+	fireEvent.press( backButton[ backButton.length - 1 ] );
+
+	// Currently logs `act` warning due to https://git.io/JYYGE
+	await waitForElementToBeRemoved( () =>
+		screen.getByText(
+			'Each block has its own settings. To find them, tap on a block. Its settings will appear on the toolbar at the bottom of the screen.'
+		)
+	);
+} );
```

Comment:
Probably you could use the `waitFor` helper we used on other integration tests to prevent the warning:
https://github.com/WordPress/gutenberg/blob/8e4ebfae09351aa5d21fe072deb16d265bdb130e/test/native/helpers.js#L67-L101

Reply:
The challenge with this particular assertion is that we are asserting its value is `null`. If we... 

```js
await waitFor(() => elementCurrentlyVisibleButShouldGoAway)
```

...the test continues forward immediately (and fails). If we...

```js
await waitFor(() => expect().toBeNull())
```

...the test never passes because I believe `toBeNull` doesn't return anything. 

I believe this is likely a shortcoming of our internal `waitFor` implementation, where we [expect a truthy value](https://github.com/WordPress/gutenberg/blob/acf909f8f4a15db031fb5dce3979d7f8556f4162/test/native/helpers.js#L79). So, there is likely room to implement a custom `waitForElementToBeRemoved` or refactor our `waitFor` to not check the presence of a truthy value. E.g. it would appear RNTL's `waitFor` merely [ensures the expectation does not throw an error](https://github.com/callstack/react-native-testing-library/blob/54d970c94013966c8839ee318b99737d64296d83/src/waitFor.js#L138-L167). 

However, I plan to merge this work as-is as I am limited on time currently. We can hopefully improve the error log in this test at a later point. Let me know if you think my understanding of our internal `waitFor` makes sense.

Reply:
Thanks for clarifying it ❤️ ! Interesting, I didn't realize about this particularity of asserting `null`.

> I believe this is likely a shortcoming of our internal waitFor implementation, where we expect a truthy value. So, there is likely room to implement a custom waitForElementToBeRemoved or refactor our waitFor to not check the presence of a truthy value. E.g. it would appear RNTL's waitFor merely ensures the expectation does not throw an error.

Right, `waitFor` won't work for this case as it's always expecting the result to be truthy. I think this behavior makes sense since it's expected that the callback passed to the function returns a value. I'd lean towards implementing a custom `waitForElementToBeRemoved` which probably would reuse most of the code we have for `waitFor`.

> However, I plan to merge this work as-is as I am limited on time currently. We can hopefully improve the error log in this test at a later point. Let me know if you think my understanding of our internal waitFor makes sense.

Agreed, the purpose of this comment was just to capture insights about the use of `waitFor`, so please feel free to merge it :shipit:.